doc/developer-notes.md

```diff
@@ -40,11 +40,23 @@ class Class
 }
 ```
 
-- Strict formatting is enforced all files listed in "src/.formatted-files".
+- Strict formatting is enforced for all files listed with the "filter=bitcoin-clang-format"
+  attribute in "src/.gitattributes".
   Your pull request will fail the Travis build if one of these files is not formatted properly
 
 - To auto-format a file:
 
+  The fully-automatic way:
+  ========================
+  Add the following to your $HOME/.gitconfig or .git/config in the Bitcoin directory:
+  ```
+[filter "bitcoin-clang-format"]
+        clean = "clang-format-3.8 -style=file %f|contrib/devtools/trailing-comment.py 120"
```

Comment:
trailing-comment.py needs to run before clang-format.  But best of all, call contrib/devtools/clang-format.py.  Using this script will ensure that your code always matches make check-formatting...